src/magic_folder/magic_folder.py

```diff
@@ -918,3 +962,92 @@ def add_file(self, path):
+            self._tahoe_client,
+        )
+        self._processing_loop.clock = self._clock
+        self._processing = self._processing_loop.start(self._polling_interval, now=True)
+
+    def stopService(self):
+        """
+        Stop the uploader service.
+        """
+        service.Service.stopService(self)
+        d = self._processing
+        self._processing_loop.stop()
+        self._processing = None
+        self._processing_loop = None
+        return d
+
+
+    @eliotutil.inline_callbacks
+    def _upload_local_snapshots(self, tahoe_client):
+        """
+        Check the db for uncommitted LocalSnapshots, deserialize them from the on-disk
+        format to LocalSnapshot objects and commit them into the grid.
+
+        :param tahoe_client: a TahoeClient instance
+        """
+
+        # get the mangled paths for the LocalSnapshot objects in the db
+        localsnapshot_names = self._snapshot_store.get_all_item_paths()
+
+        # XXX: processing this table should be atomic. i.e. While the upload is
+        # in progress, a new snapshot can be created on a file we already uploaded
+        # but not removed from the db and if it gets removed from the table later,
+        # the new snapshot gets lost. Perhaps this can be solved by storing each
+        # LocalSnapshot in its own row than storing everything in a blob?
+        # https://github.com/LeastAuthority/magic-folder/issues/197
+        for name in localsnapshot_names:
+            action = UPLOADER_SERVICE_UPLOAD_LOCAL_SNAPSHOTS(relpath=name)
+            with action:
+                # deserialize into LocalSnapshot
+                snapshot = self._snapshot_store.get_local_snapshot(name, self._local_author)
+
+                # now upload each item in the queue
+                try:
+                    remote_snapshot = yield write_snapshot_to_tahoe(
+                        snapshot,
+                        self._local_author,
+                        tahoe_client,
+                    )
+
+                    # At this point, remote snapshot creation successful for
+                    # the given relpath. Remove the LocalSnapshot from the db.
+                    yield self._snapshot_store.remove_localsnapshot(name)
+
+                    # store the remote snapshot capability in the db.
+                    yield self._snapshot_store.store_remote_snapshot(name, remote_snapshot)
+
+                except NoServersError:
+                    # Unable to reach Tahoe storage nodes because of
+                    # network errors or because the tahoe storage nodes
+                    # are offline. Retry?
+                    # XXX: Perhaps implement exponential backoff for retry?
+                    NO_NETWORK.log()
+                    continue
+                except Exception:
+                    # all other exceptions, pass on upstream
+                    proc = Failure()
+                    write_failure(proc)
```

Comment:
If you let these exceptions propagate through the `with action` block then Eliot will automatically mark the action a failure and record some details about the failure.  That might be sufficient for our purposes.  You still have to catch the exception outside the `with action` block to stop it propagating too far, but you may not have to _do_ anything with it there (certainly you don't have to log it again).

Reply:
Would doing the following be equivalent then? It would log the exception and the failed action, which seem to be what we want here. 
```
...
except Exception:
    write_traceback()
```